sdk/identity/Azure.Identity/README.md

```diff
@@ -1,99 +1,175 @@
-### Authenticate the client
+## DefaultAzureCredential
+`DefaultAzureCredential` is appropriate for most scenarios where the application is intended to run in the Azure Cloud. This is because the `DefaultAzureCredential` determines the appropriate credential type based of the environment it is executing in. It supports authenticating both as a service principal or managed identity, and can be configured so that it will work both in a local development environment or when deployed to the cloud. 
 
-If your library requires authentication for use, such as for Azure services, include instructions and example code needed for initializing and authenticating.
+The `DefaultAzureCredential` will first attempt to authenticate using credentials provided in the environment. In a development environment you can authenticate as a service principal with the `DefaultAzureCredential` by providing configuration in environment variables as described in the next section.
 
-For example, include details on obtaining an account key and endpoint URI, setting environment variables for each, and initializing the client object.
+If the environment configuration is not present or incomplete, the `DefaultAzureCredential` will then determine if a managed identity is available in the current environment.  Authenticating as a managed identity requires no configuration, but does
+require platform support. See the
+[managed identity documentation](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/services-support-managed-identities) for more details on this.
 
-## Key concepts
+## Environment variables
 
-The *Key concepts* section should describe the functionality of the main classes. Point out the most important and useful classes in the package (with links to their reference pages) and explain how those classes work together. Feel free to use bulleted lists, tables, code blocks, or even diagrams for clarity.
+`DefaultAzureCredential` and `EnvironmentCredential` are configured for service
+principal authentication with these environment variables:
 
-## Examples
+|variable name|value
+|-|-
+|`AZURE_CLIENT_ID`|service principal's app id
+|`AZURE_TENANT_ID`|id of the principal's Azure Active Directory tenant
+|`AZURE_CLIENT_SECRET`|one of the service principal's client secrets
 
-Include code snippets and short descriptions for each task you listed in the [Introduction](#introduction) (the bulleted list). Briefly explain each operation, but include enough clarity to explain complex or otherwise tricky operations.
 
-If possible, use the same example snippets that your in-code documentation uses. For example, use the snippets in your `examples.py` that Sphinx ingests via its [literalinclude](https://www.sphinx-doc.org/en/1.5/markup/code.html?highlight=code%20examples#includes) directive. The `examples.py` file containing the snippets should reside alongside your package's code, and should be tested in an automated fashion.
+## Examples
 
```

Comment:
It will be good to clarify that for the purposes of these examples you are going to use another Azure Client Library, the Azure.Security.KeyVault.Secrets and a link to it so people can easily make the connection

Reply:
Before every example when u change clients actually